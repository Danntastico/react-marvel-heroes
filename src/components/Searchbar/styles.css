.herosearch__searchbar {
  position: absolute;
  left: 20px;
  bottom: 0;
  top: 130px;
  z-index: 1;
  transform: translate(0%, -50%);
  height: 40px;
  /* padding: 10px; */
  display: flex;
  transition: 0.4s;
}
.herosearch__searchbar--active {
  opacity: 0.7;
}
.herosearch__searchbar--active > .search__input {
  width: 150px;
  padding-left: 3px;
}
.herosearch__searchbar--active:hover {
  opacity: 1;
}

.herosearch__searchbar:hover > .search__input {
  width: 150px;
  padding: 0 20px;
}
.herosearch__searchbar:hover > .search__btn {
  border-radius: 10px;
  font-size: 20px;
}
.search__btn {
  text-decoration: none;
  display: flex;
  justify-content: center;
  align-items: center;
  transition: 0.4s;
  color: white;
}

.search__input {
  border: none;
  background: none;
  outline: none;
  font-weight: 100;
  font-size: 0.9em;
  width: 0px;
  transition: 0.4s;
  color: white;
}
.search__input::placeholder {
  color: white;
  font-weight: lighter;
  font-family: 'Roboto', sans-serif;
}
