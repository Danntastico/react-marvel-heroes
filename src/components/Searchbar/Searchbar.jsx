import React from 'react';
import { useForm } from '../../hooks/useForm';

import './styles.css';

export const Searchbar = ({ setHeroByName }) => {
  const [{ description }, handleInputChange] = useForm({
    description: '',
  });

  const handleSubmit = (e) => {
    e.preventDefault();
    setHeroByName(description);
  };
  const activeInput =
    description !== '' ? 'herosearch__searchbar--active' : ' ';
  return (
    <form
      className={`herosearch__searchbar ${activeInput}`}
      onSubmit={handleSubmit}
    >
      <input
        type='text'
        name='description'
        placeholder='Search...'
        className='search__input'
        autoComplete='off'
        value={description}
        onChange={handleInputChange}
      />
      <div className='search__btn'>
        <i className='fas fa-search'></i>
      </div>
    </form>
  );
};
