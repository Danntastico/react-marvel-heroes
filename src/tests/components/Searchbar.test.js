import React from 'react';
import { shallow } from 'enzyme';
import { Searchbar } from '../../components/Searchbar/Searchbar';

import '@testing-library/jest-dom';

describe('Unit test on <SearchBar />', () => {
  const setHeroByName = jest.fn();
  let wrapper = shallow(<Searchbar setHeroByName={setHeroByName} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<Searchbar setHeroByName={setHeroByName} />);
  });
  test('Should show correctly (shallow rendering)', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Should call setHeroByName', () => {
    const value = 'iron man';
    wrapper.find('input').simulate('change', { target: { value } });

    wrapper.find('form').simulate('submit', { preventDefault() {} });

    expect(setHeroByName).toHaveBeenCalled();
    expect(setHeroByName).toHaveBeenCalledTimes(1);
  });
});
