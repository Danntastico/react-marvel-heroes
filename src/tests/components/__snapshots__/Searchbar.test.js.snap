// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test on <SearchBar /> Should show correctly (shallow rendering) 1`] = `
<form
  className="herosearch__searchbar data-card  "
  onSubmit={[Function]}
>
  <input
    autoComplete="off"
    className="search__input"
    name="description"
    onChange={[Function]}
    placeholder="Search..."
    type="text"
    value=""
  />
  <div
    className="search__btn"
  >
    <i
      className="fas fa-search"
    />
  </div>
</form>
`;
